src/telemetry/client.cpp

```diff
@@ -108,36 +110,23 @@ std::string Client::receiveData()
   char header[8];
 
   // receive header
-  if (recv(sockfd_, header, 8, 0) == -1) {
-    log_.ERR("Telemetry", "Error receiving header");
-    throw std::runtime_error{"Error receiving header"};  // NOLINT
+  if (recv(sockfd_, header, 8, 0) <= 0) {
+    throw std::runtime_error{"Error receiving header"};
   }
 
   int payload_length = strtol(header, NULL, 0);
   char buffer[1024];  // power of 2 because apparently it's better for networking
   memset(buffer, 0, sizeof(buffer));  // fill with 0's so null terminated by default
 
   // receive payload
-  if (recv(sockfd_, buffer, payload_length, 0) == -1) {
-    log_.ERR("Telemetry", "Error receiving payload");
-    throw std::runtime_error{"Error receiving payload"};  // NOLINT
+  if (recv(sockfd_, buffer, payload_length, 0) <= 0) {
+    throw std::runtime_error{"Error receiving payload"};
```

Comment:
My understanding of this is not very good. So just to make sure: Is `<=` the right choice here? What is `recv` returning?

Reply:
`-1` is returned when there is an error and `0` is returned when the client disconnects. So `<=` allows to handle both.

> Upon successful completion, `recv()` shall return the length of the message in bytes. If no messages are available to be received and the peer has performed an orderly shutdown, `recv()` shall return 0. Otherwise, -1 shall be returned and `errno` set to indicate the error.
From https://pubs.opengroup.org/onlinepubs/9699919799/functions/recv.html

Reply:
Alright, cheers! Maybe add a little comment to note that?